### PR TITLE
fix(switch): make Switch a controlled component

### PR DESCRIPTION
**Pull request by Mcourt**
# General summary

Fix Switch component to make it a stateless controlled component

# Screenshots

lumapps_ds_controllable_switch

# Check list

-   [ ] (if has react) Check through the [react dev check list]
-   [ ] (if has style) Request an integration review
-   [x] (if has docs) Request a doc review
-   [x] Request a frontend review
-   [x] (if fix or feature) Request a test

Check out the [contribution guide] for general guidelines for submissions to the design system.

[contribution guide]: https://github.com/lumapps/design-system/blob/master/CONTRIBUTING.md
[react dev check list]: https://github.com/lumapps/design-system/blob/master/CONTRIBUTING.md#react-developpment-check-list

### CODE REVIEW

File: CHANGELOG.react.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 -   Fix stroke-width property on react progress bar.
 -   Fixes to Text Area related to Input resizing and `valid` and `error` states in Material
+-   Make `Switch` a controlled component
```
Comment (gcornut) [inline CHANGELOG.react.md]: This is a breaking change even though no one might be using this component right now ^^.
We usually mark breaking change with `_[BREAKING]_`

Comment (Mcourt) [reply CHANGELOG.react.md]: True ! Will update.

**Comment by FlorentDinet** [comment]
Doc and Demo are ok but the **dark background switch** of the demo block is broken (side-effect I guess)

Oct-25-2019 15-19-04

**Comment by gcornut** [comment]
> Doc and Demo are ok but the **dark background switch** of the demo block is broken (side-effect I guess)

Nice catch, the `DemoBlock` component needs to be updated

**Comment by Mcourt** [comment]
@FlorentDinet @gcornut Nice catch indeed ! Fixed :)

**Comment by Mcourt** [comment]
> Could your remove the `demo/react/components/switch` folder?
> Sorry, last thing I ask ^^

Done :)

**Comment by FlorentDinet** [comment]
👌

**Comment by gcornut** [comment]
@Mcourt Watch out when you'll rebase this branch onto master, your modifications in the CHANGELOG might appear under v0.12 but it should appear in under `Unreleased`

**Comment by Mcourt** [comment]
> @Mcourt Watch out when you'll rebase this branch onto master, your modifications in the CHANGELOG might appear under v0.12 but it should appear in under `Unreleased`

@gcornut I thought I wasn't supposed to merge my own Design System PR's :sweat_smile:

**Comment by Mcourt** [comment]
> @Mcourt Watch out when you'll rebase this branch onto master, your modifications in the CHANGELOG might appear under v0.12 but it should appear in under `Unreleased`

@gcornut Good call !

**Pull request merged**